### DISCUSSION TITLE
Four (now five!) data science-type type books for us to use

### DISCUSSION THREAD

**Discussion by rdstern** [IDEMSInternational/R-Instat#6598] [General]
These are all available freely to read.  Printed copies would be bought.  The online versions have the advantage that R code can be copied and pasted into RStudio or R-Instat to try.

The Wickham book on [R for Data Science](https://r4ds.had.co.nz/index.html), has been our "standard".  It starts, as we propose, on data visualisation.

The book called [R Graphics cookbook](https://r-graphics.org/) is by Winsten Chang, who also works at RStudio.  It is more general that appears from the title and also discusses data science from the start.  

The third is called [Introduction to Data Science](https://rafalab.github.io/dsbook/).  The author is Rafael A. Irizarry and is seems to be a book used at Harvard.  There is also a useful R package with all the datasets used in this book.  It provides a good model for the topics we aim to cover before too long.  

The fourth is again narrower and is just called [Data Visualisation with R](https://rkabacoff.github.io/datavis/).  It is particularly interesting in that it has chapters in the same order as we approach the data visualisation in R-Instat, namely 1 variable, (Chapter 3), then 2 variables (Chapter 4) and so on.  That's what we want to do in our data visualisation too, i.e. it would be good if our Describe > One Variable and Describe > Two Variable dialogues offered most (if not all) of the graph types that are in these chapters of the book.

We should mix using the same exam[ples as in these books and trying out the same ideas on our own data sets.

**Comment by rdstern** [comment]
I have taken one of these books - just the start of it.  Written the following document:
[Data Visualization with R_Instat.docx](https://github.com/africanmathsinitiative/R-Instat/files/7090414/Data.Visualization.with.R_Instat.docx)

This document describes 5 ways of visualising data using R-Instat.  It uses an example from chapter 3 of the data visualisation book.  The next steps are to ensure that  most (ideally all) the descriptive visualisations in this book can be done easily in R-Instat.

**Reply by rachelkg** [reply]
I was starting work on the  R Graphics cookbook data visualisation chapters. Attempting to recreate items from these in R-Insat. Should I work through this first instead?

**Comment by rachelkg** [comment]
I have tried to recreate the graph from Chapter 7 https://rafalab.github.io/dsbook/ggplot2.html. Introduction to Data Science. 

I had some success and some challenges. Feel free to make comments and improvements. 

[Trying to recreate the Dslabs graph on murders.docx](https://github.com/africanmathsinitiative/R-Instat/files/7172187/Trying.to.recreate.the.Dslabs.graph.on.murders.docx)
 have documented them in this word document.

**Comment by rdstern** [comment]
It is time for considerable improvements in our modelling menu.  I will be suggesting a set of issues.  I also use another recent book as support.  It is called [Introduction to Statistical Learning with applications in R](https://hastie.su.domains/ISLR2/ISLRv2_website.pdf).  It is the second edition and is accompanied by a package with 21 (I think) interesting examples of data.  Each chapter includes a section on worked examples with these datasets - of course using R.  So our challenge is for (most of) these models to be available for users, ideally via dialogues, and at least through the halfway dialogues.
